mdsuite/transformations/map_molecules.py

```diff
@@ -240,20 +239,38 @@ def _update_species_type_dict(
 
     def _build_reference_graphs(self):
         """
-        Build the reference graphs from the SMILES strings.
+        Build the reference graphs from the reference input.
+        .
         Returns
         -------
-        Nothing.
+        Updates the reference molecules attribute of the class. It should be populated
+        after this call and look something like:
+
+        {'bmim': {'species': ['C', 'H', 'N'], 'mass': 200, 'graph': tf.Tensor}}
+
+        If a dict is used as a reference, the graph will not exist and stronger
+        isomorphism checks will not be possible.
         """
         for item in self.molecules:
             self.reference_molecules[item] = {}
-            mol, species = MolecularGraph(
-                self.experiment,
-                from_smiles=True,
-                smiles_string=self.molecules[item]["smiles"],
-            ).build_smiles_graph()
+            if 'smiles' in self.molecules[item]:
+                mol, species = MolecularGraph(
+                    self.experiment,
+                    from_smiles=True,
+                    smiles_string=self.molecules[item]["smiles"],
+                ).build_smiles_graph()
+                self.reference_molecules[item]["graph"] = mol
+            elif 'reference' in self.molecules[item]:
+                species = self.molecules[item]['reference']
+            else:
+                error_msg = "The minimum amount of data was not given to the mapping." \
+                            "Either provide a reference key with information about" \
+                            "Which species and the number of them are in the molecule," \
+                            "or provide a SMILES string that can be used to compute " \
+                            "this information."
+                log.info(error_msg)
+                raise ValueError("Provide either a reference key or smiles string")
```

Comment:
I think we don't need the logging here or make it `log.critical` because it fails afterwards

Reply:
Yes I will move the message into the exception and remove the log or make it critical, I guess it doesn't hurt to have it logged somewhere.